### ISSUE TITLE
Muon Front End: Test Barndoors and momentum slits

### ISSUE THREAD

**Issue by John-Holt-Tessella** [ISISComputingGroup/IBEX#1424]
Barndoors and momentum slits need to be tested on an instrument.

Ticket #1388 implemented barndoors and momentum slits but they need to be tested on an actual machine.

Set home vals to 0 see #2471.

**Comment by John-Holt-Tessella** [comment]
Electrical now done waiting for an opportunity to test

**Comment by KathrynBaker** [comment]
Have checked, and the electrical is working and in place for the barn doors, so that it is possible to operate all four sets of slits from the same Galil with encoders.  Waiting on the removal of vac grease and permission to move the momentum slits (Shutdown in November is next likely opportunity).

We should consider doing this in the following way:
Ensure that the slit control on each of the NDX systems is turned off
Setup the NDE to control the slit sets

In the first instance, the quick and easy implementation for a two week shutdown:
Update the LabVIEW code for the slits to use a different thread for EMU (move it from thread 1 to thread 4 as thread 1 is used by the IOC)
Ensure that only the momentum slits are being initialised on the NDE
Start the IOC, make sure that the momentum slits move and behave as expected
Start the slits on each NDX and make sure that everything can run together as expected

If there is time the following should be tested, and if it works this setup can be used after this shutdown (don't forget the LabVIEW setup), if it fails, the rework and final implementation should be delayed until a longer shut down:
Ensure that the LabVIEW is stopped
Set the NDE to use the barndoors as well
Try the code and see if it all works as expected

**Comment by KathrynBaker** [comment]
Impeded as awaiting access to the system which is outside of our control, this was meant to be in the November 2016 shutdown, but there may be a delay. When the equipment is available this will become a matter of urgency, so keeping it in impeded until certain that this will be delayed until 2017.

**Comment by kjwoodsISIS** [comment]
Dependent on motors being ready.  This will be an "awaiting" ticket.

**Comment by KathrynBaker** [comment]
John and I have had a bit of a look at this, and at the moment there needs to be some work and research to check how (and possibly if) it will work.

The current plan involves:
- [ ] seeing if we can find out if, and therefore where analog inputs 1,2,3 of the Galil are read
- [ ] seeing if the linking to a different readback record will work
- [ ] seeing if the motor can be moved according to that readback value
- [ ] if we can't use the motor record, seeing if we can find a way to use the existing Galil programme via IBEX

**Comment by KathrynBaker** [comment]
After a productive conversation, Freddie and I have come to the conclusion that we will have to use the 4th option in the previous comment, and use the program in the Galil. As such there is going to be a new option put in place for the Galil under a new ticket which will allow for alternative methods of setting positions to the standard Galil commands #1990. It is also likely to be helpful for other future work such as Oscillating Radial Collimators.

As we are only able to experiment with these out of cycle, and this now needs a driver change, I shall put this into impeded, as soon as the driver change has been made we can update NDEMUONFE with it, and return to this. Should there be issues with that driver change, and this is still impeded at the start of cycle, then it can be moved to the backlog, and once the driver is changed be moved back to awaiting.

**Comment by KathrynBaker** [comment]
As the program for the Galil is going to remain, one change that has been requested by the operations team is that the limit switches are honoured, this will mean a fairly minor code change to a galil program, as such I shall include it under this ticket

**Comment by KathrynBaker** [comment]
The PV names in the BARNDOORS should also be shortened, and renamed to SLITS, as this is more meaningful to the Scientists.

**Comment by KathrynBaker** [comment]
Due to a mechanical failure and proximity to cycle start, this is now going to have to wait for a new opportunity,  as such moving to awaiting as the next opportunity may be a beam off day, or the next shutdown.

**Comment by KathrynBaker** [comment]
Don't forget to add the homing method for the momentum slits

**Comment by KathrynBaker** [comment]
Update January 2018 - this is still awaiting a suitable opportunity, each time it can be planned to be done, there has been a hardware issue (as has just occurred) which makes the change to the software a risk too many for running during the cycle.

**Comment by KathrynBaker** [comment]
Update March 2018 - This will happen after cycle 2018/01

**Comment by KathrynBaker** [comment]
These still don't work, and as such will need to be revisited, see ticket #3196 for the steps to undertake before retesting these.

**Comment by KathrynBaker** [comment]
Slits being replaced, so not undertaking this task
